### DISCUSSION TITLE
How can I add new robot to OmniGibson like Kinova Gen3?

### DISCUSSION THREAD

**Discussion by DawnyWu** [StanfordVL/BEHAVIOR-1K#982] [Q&A]
How can I add new robot to OmniGibson like Kinova Gen3?

**Comment by wensi-ai** [comment]
Hi,

Thank you for your interest in OmniGibson. We have a tutorial for importing custom robots [here](https://behavior.stanford.edu/omnigibson/tutorials/custom_robot_import.html
)

In general, you will need to prepare a USD file for your custom robot (which can be converted from URDFs), then write a robot class under `omnigibson/robots`. You can test the robot with the `robot_control_example` we provide. The recommended workflow will be to first create a minimum version of the robot class with just null controllers to test the USD can be successfully imported, then gradually add control functionalities to make the robot actually controllable.
